src/payment/payment-methods/stripe/service.js

```diff
@@ -17,12 +17,53 @@ ensureConfig(['ECOMMERCE_BASE_URL', 'STRIPE_RESPONSE_URL'], 'Stripe API service'
  */
 export default async function checkout(
   basket,
-  { paymentIntentId, skus },
+  { skus, elements, stripe, context, values },
   setLocation = href => { global.location.href = href; }, // HACK: allow tests to mock setting location
 ) {
+  const {
+    firstName,
+    lastName,
+    address,
+    unit,
+    city,
+    country,
+    state,
+    postalCode,
+    organization,
+    purchasedForOrganization,
+  } = values;
+
+  const result = await stripe.updatePaymentIntent({
+    elements,
+    params: {
+      payment_method_data: {
+        billing_details: {
+          address: {
+            city,
+            country,
+            line1: address,
+            line2: unit || '',
+            postal_code: postalCode || '',
+            state: state || '',
+          },
+          email: context.authenticatedUser.email,
+          name: `${firstName} ${lastName}`,
+        },
+        metadata: {
+          organization,
+          purchased_for_organization: purchasedForOrganization,
+        },
+      },
+    },
+  });
+
+  if (result.error?.code === 'payment_intent_unexpected_state' && result.error?.type === 'invalid_request_error') {
+    handleApiError(result.error);
+  }
+
```

Comment:
Since the error handling was removed from calling Stripe's updatePaymentIntent for `card_error` `validation_error` and `else` other than the ones you've added on L60, we don't need the `setMessage` local state anymore, could you remove it from `StripePaymentForm`? Also assuming that the cases that were removed were tested manually?

Reply:
confirming here for posterity that card error and validation error will no longer show text underneath the stripe form when Place Order is pressed. they will have the red text on the field itself, and additionally, a red banner with a generic error will be displayed (in addition to the red text) if a user tries to place an order with a card/validation error